imap_processing/codice/tests/test_decom.py

```
@@ -0,0 +1,104 @@
+"""Tests the decommutation process for CoDICE CCSDS Packets"""
+
+from pathlib import Path
+
+import pandas as pd
+import pytest
+
+from imap_processing import imap_module_directory
+from imap_processing.decom import decom_packets
+
+
+@pytest.fixture(scope="session")
+def decom_test_data():
+    """Read test data from file"""
+    packet_file = Path(
+        f"{imap_module_directory}/codice/tests/data/"
+        f"raw_ccsds_20230822_122700Z_idle.bin"
+    )
+    xtce_document = Path(
+        f"{imap_module_directory}/codice/packet_definitions/P_COD_NHK.xml"
+    )
+    data_packet_list = decom_packets(packet_file, xtce_document)
+    data_packet_list = [
+        packet
+        for packet in data_packet_list
+        if packet.header["PKT_APID"].raw_value == 1136
+    ]
+
+    return data_packet_list
+
+
+@pytest.fixture(scope="session")
+def validation_data():
+    """Read in validation data from the CSV file
+
+    Returns
+    -------
+    validation_data : pandas DataFrame
+        The validation data read from the CSV, cleaned up and ready to compare
+        the decommutated packet with
+    """
+
+    # Read in the CSV file (perhaps to a pandas dataframe?)
+    validation_file = Path(
+        f"{imap_module_directory}/codice/tests/data/"
+        f"idle_export_raw.COD_NHK_20230822_122700.csv"
+    )
+    validation_data = pd.read_csv(validation_file)
+    # Remove the timestamp column and data
+    if "timestamp" in validation_data.columns:
+        validation_data = validation_data.drop(columns=["timestamp"])
```

Nice, glad this was as easy thing to do!

---

Do you need to drop the timestamp? Or can you just ignore it later in your processing if it isn't used there.

Also, I think you can do this all in one line and ignore the errors if that column doesn't exist.
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.drop.html

```python
validation_data.drop(columns=["timestamp"], inplace=True, errors="ignore")
```